mozdownload/scraper.py

```diff
@@ -972,41 +972,43 @@ def get_build_info_for_index(self, build_index=None):
         """Get additional information for the build at the given index."""
         url = urljoin(self.base_url, self.build_list_regex)
 
-        self.logger.info('Retrieving list of builds from %s' % url)
-        parser = self._create_directory_parser(url)
-        parser.entries = parser.filter(r'^\d+$')
+        if not self.timestamp:
```

Comment:
Would you mind moving the case with a timestamp up here? It's a personal taste but having less complex conditions including not-negative ones make the code easier to read and follow. Here it's not a problem but it would be good to follow that pattern.